pkg/inject/inject.go

```diff
@@ -187,12 +192,19 @@ func (conf *ResourceConfig) ParseMetaAndYAML(bytes []byte) (*Report, error) {
 	return newReport(conf), nil
 }
 
-// GetPatch returns the JSON patch containing the proxy and init containers specs, if any
-func (conf *ResourceConfig) GetPatch(bytes []byte) (*Patch, error) {
+// GetPatch returns the JSON patch containing the proxy and init containers specs, if any.
+// If injectProxy is false, only the config.linkerd.io annotations are set.
+func (conf *ResourceConfig) GetPatch(bytes []byte, injectProxy bool) (*Patch, error) {
 	patch := NewPatch(conf.workload.metaType.Kind)
 	if conf.pod.spec != nil {
-		conf.injectObjectMeta(patch)
-		conf.injectPodSpec(patch)
+		if len(conf.pod.meta.Annotations) == 0 {
+			patch.addPodAnnotationsRoot()
+		}
+		if injectProxy {
+			conf.injectObjectMeta(patch)
+			conf.injectPodSpec(patch)
+		}
+		conf.injectPodAnnotations(patch)
```

Comment:
I think the call to `inject.injectPodAnnotations()` needs to happen earlier, because it populates the `conf.pod.meta.Annotations` map used to override the configs in `conf.injectPodSpec()`.

Reply:
Good point :+1:

Reply:
This also means there's a gap in the unit tests to be filled :mag:

Reply:
There were no tests for CLI option overrides. I augmented inject_test.go to take that into account and that did catch the problem you alluded to.